internal/validator/cosignvalidator/cosign_validator.go

```diff
@@ -264,26 +291,75 @@ func (cv *CosignValidator) ValidateImage(
 		return k, nil
 	}
 
+	// propagate errors if no digests were found
+	if len(propagateErrors) > 0 {
+		return "", fmt.Errorf("error validating image: %s", propagateErrors)
+	}
+
 	// no signed digests
 	return "", fmt.Errorf("no signed digests")
 }
 
+func (cv *CosignValidator) validateWithVerifier(
+	ctx context.Context,
+	imageRef name.Reference,
+	opts *cosign.CheckOpts,
+	verifier Verifier,
+	verifierOutputChannel chan<- verifierOutput,
+) {
+	copts := *opts
```

Comment:
Just from an API point, I think, it would be nicer to handover the value instead of pointer, such that it is clear that the original opts shouldn't (and factually) cannot be changed by the goroutine